pennylane/templates/subroutines/qchem/basis_rotation.py

```diff
@@ -124,10 +134,10 @@ class BasisRotation(Operation):
         >>> from scipy.stats import ortho_group
         >>> O = ortho_group.rvs(4, random_state=51)
         >>> print(qml.draw(qml.BasisRotation(wires=range(4), unitary_matrix=O).decomposition)())
-        0: ──Rϕ(3.14)─╭G(-3.19)──────────╭G(2.63)─┤
-        1: ─╭G(-3.13)─╰G(-3.19)─╭G(2.68)─╰G(2.63)─┤
-        2: ─╰G(-3.13)─╭G(-2.98)─╰G(2.68)─╭G(5.70)─┤
-        3: ───────────╰G(-2.98)──────────╰G(5.70)─┤
+        0: ──Rϕ(3.14+0.00j)─╭G(-3.19)──────────╭G(2.63)─┤
```

Comment:
Sorry for only realizing this now, but I think it would be neat if we can avoid passing complex numbers into `PhaseShift` here.
For the real-valued (or close-to-real-valued) case, the determinant only ever is `1` or `-1`, only ever requiring a phase shift of `π`or `0`. 
So we could do `np.pi * (1- math.real(det))/2` and should be fine? 🤔